Library/Homebrew/test/formula_creator_spec.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+require "formula_creator"
+
+describe Homebrew::FormulaCreator do
+  it "parses names from URL correctly" do
+    t = described_class.name_from_url("https://github.com/abitrolly/lapce/archive/v0.3.0.tar.gz")
+    expect(t).to eq("lapce")
+    t = described_class.name_from_url("http://www.codesrc.com/gitweb/index.cgi?p=libzipper.git;a=summary")
+    expect(t).to eq("libzipper")
+    t = described_class.name_from_url("https://github.com/abitrolly/lapce.git")
+    expect(t).to eq("lapce")
+    t = described_class.name_from_url("https://github.com/stella-emu/stella/releases/download/6.7/stella-6.7-src.tar.xz")
+    expect(t).to eq("stella")
+    t = described_class.name_from_url("http://digit-labs.org/files/tools/synscan/releases/synscan-5.02.tar.gz")
+    expect(t).to eq("synscan")
+  end
```

Comment:
Please split this into one URL/expectation per test

Reply:
Do you mean split it into five `it "parses URL ..."` blocks ?

Reply:
yes